Seperate student name column into fore- and surname in the submission…

duaanakshbandi opened the pull request:
This PR fixes issue https://github.com/Sapphire-CM/Sapphire/issues/574

matthee commented on app/helpers/accounts_helper.rb:
```diff
@@ -21,6 +21,14 @@ def account_fullname_with_matriculation_number(account)
     "#{account.fullname} (#{account.matriculation_number})"
   end
 
+  def account_forname(account)
+    "#{account.forename}"
+  end
+
+  def account_surname(account)
+    "#{account.surname}"
+  end
+
```
I think these helper methods are not necessary, since they do not provide any benefits over just calling `account.forename` and `account.surname` respectively
```suggestion
```

matthee commented on app/helpers/submissions_helper.rb:
```diff
@@ -51,6 +51,18 @@ def submission_solitary_author(submission)
   end
 
+  def submission_solitary_author_forename(submission)
+    submission.students.map do |student|
+      account_forname(student)
+    end.join(", ").presence || content_tag(:em, 'unknown author')
```
This should be a one-liner:

```suggestion
    submission.students.map(&:forename).keep_if(&:present?).join(", ").presence || content_tag(:em, 'unknown author')
```

Also, I am not sure if it is even possible that accounts are missing the `forename` and `surname` attribute. Please investigate. If my hunch is correct, you could do the following:

```suggestion
    submission.students.map(&:forename).join(", ").presence || content_tag(:em, 'unknown author')
```

matthee commented on app/helpers/submissions_helper.rb:
```diff
@@ -51,6 +51,18 @@ def submission_solitary_author(submission)
+  end
+
+  def submission_solitary_author_surname(submission)
+    submission.students.map do |student|
+      account_surname(student)
+    end.join(", ").presence || content_tag(:em, 'unknown author')
```
See comment above.

```suggestion
    submission.students.map(&:surname).join(", ").presence || content_tag(:em, 'unknown author')
```

matthee commented on app/helpers/submissions_helper.rb:
```diff
@@ -51,6 +51,14 @@ def submission_solitary_author(submission)
+    submission.students.map(&:forename).join(", ").presence || content_tag(:em, 'unknown author')
+  end
+
+  def submission_author_surname(submission)
```
This method's name should be pluralized.

```suggestion
  def submission_author_surnames(submission)
```

matthee commented on app/views/staff_submissions/_submissions_list.html.erb:
```diff
@@ -20,7 +21,10 @@
+          <%= submission_author_forenames submission %>
+        </td>
+        <td>
+          <%= submission_author_surname submission %>
```
This method's name should be pluralized.

```suggestion
          <%= submission_author_surnames submission %>
```

matthee reviewed (changes requested):
Hi @duaanakshbandi. What's the status on this PR? 

Quick note: We should be consistent in naming our methods. We should either use singularized names or pluralized names. 

Also: We need tests to verify the behavior of this code. Do you need help implement them?

duaanakshbandi commented:
Dear Matthias,

Thank you very much for letting me know where to add the test, as for the current status I added the test and renamed the methods, I hope they are good now, please let me know if they need change!

matthee commented:
Great! Thank you for your work! LMGTM
